services/venice-server/src/main/java/com/linkedin/venice/server/VeniceServer.java

```diff
@@ -310,14 +312,20 @@ private List<AbstractVeniceService> createServices() {
         ? new RocksDBMemoryStats(metricsRepository, "RocksDBMemoryStats", plainTableEnabled)
         : null;
 
+    boolean whetherToRestoreDataPartitions = !isIsolatedIngestion()
+        || veniceConfigLoader.getVeniceServerConfig().freezeIngestionIfReadyToServeOrLocalDataExists();
+
     // Create and add StorageService. storeRepository will be populated by StorageService
     storageService = new StorageService(
         veniceConfigLoader,
         storageEngineStats,
         rocksDBMemoryStats,
         storeVersionStateSerializer,
         partitionStateSerializer,
-        metadataRepo);
+        metadataRepo,
+        whetherToRestoreDataPartitions,
+        true,
+        functionToCheckWhetherStorageEngineShouldBeKeptOrNot());
```

Comment:
in the way that you've structured this PR it seems what you're doing is waiting for the managerFuture to complete as your means to do the local state clean up.  That's not a bad idea actually, I think that's actually pretty clever.  However, if you do that, then we don't need to mess with this at all.  Instead we can leave this code alone, and just use the shortened constructor that we were using before.

Reply:
(Revised).